packages/openapi-generator/src/parser/operation.ts

```diff
@@ -64,102 +59,86 @@ export function getRelevantParameters(
 ): OpenAPIV3.ParameterObject[] {
   const resolvedParameters = parameters
     .map(param => refs.resolveObject(param))
-    // Filter cookie and header parameters
+    // Filter cookie and header parameters.
     .filter(param => param.in === 'path' || param.in === 'query');
   return filterDuplicatesRight(
     resolvedParameters,
     (left, right) => left.name === right.name && left.in === right.in
   );
 }
 
-function isPlaceholder(pathPart: string): boolean {
-  return /^\{.+\}$/.test(pathPart);
-}
-
-function isValidPlaceholder(placeholder: string): boolean {
-  // This regex matches the cases:
-  // 1. it starts with `{`
-  // 2. it ends with `}`
-  // 3. it does not contain any other `{` or `}` in the middle
-  return /^\{[^{}]+\}$/.test(placeholder);
-}
-
-// This function checks whether the given path pattern is valid. Typically, it detects the invalid pattern like below
-// 1. `/path/{p1}:{p2}`
-// 2. `/path?{param}`
-function isValidPathPattern(
+/**
+ * @internal
+ */
+export function parsePathPattern(
   pathPattern: string,
-  placeholders: string[]
-): boolean {
-  return (
-    pathPattern.includes('?') ||
-    placeholders.some(placeholder => !isValidPlaceholder(placeholder))
-  );
-}
-
-function sortPathParameters(
-  pathParameters: OpenAPIV3.ParameterObject[],
-  pathPattern: string
-): OpenAPIV3.ParameterObject[] {
-  const pathParts = pathPattern.split('/');
-  const placeholders = pathParts.filter(part => isPlaceholder(part));
+  pathParameters: OpenApiParameter[]
+): string {
+  // Get the innermost curly bracket pairs with non-empty and legal content as placeholders.
+  const placeholders = pathPattern.match(/{[^/?#{}]+}/g);
+  // Get the non-parameter strings as static parts
+  const staticParts = pathPattern.split(/{[^/?#{}]+}/);
 
-  if (isValidPathPattern(pathPattern, placeholders)) {
-    throw new Error(
-      `Path pattern '${pathPattern}' is invalid or not supported.`
-    );
+  if (!placeholders) {
+    // No placeholder or path parameter found.
+    if (pathParameters.length > 0) {
+      throw new Error(
+        `Could not find placeholder for path parameter '${pathParameters
+          .map(pathParameter => pathParameter.originalName)
+          .join("', '")}'.`
+      );
```

Comment:
> 1. This will give you an error like this for multiple parameters: `Could not find placeholder for path parameter 'param1', 'param2'.`
>    Maybe I am a little too strict here, but in that case it should be plural, don't you think?

Maybe the best way to do this is to print `parameter(s)`. 😆 

> 2. If all path parameters are missing in the placeholders I will get the message above, but if 2 out of 3 are missing I will just get this message: `Could not find path parameter for placeholder 'param2'.` I find this a bit inconsistent.
> 
> Could we check this once (see below) + add tests for those two cases?

Sorry, and ignore the comment in the code. I somehow wrote it wrong.

The above if-conditions mean: **no placeholder found** && **path parameter found** => could not find placeholder for path parameter. <=> iterate through path parameters and check if there is a corresponding placeholder.

So it basically means all placeholders are missing in the path parameters, not the other way around. 

In the other direction `Could not find path parameter for placeholder` <=> iterate through placeholders and check if there is a corresponding path parameter.

I hope I am not confusing myself. 😄